ui/src/components/DenylistTable/index.jsx

```diff
@@ -0,0 +1,140 @@
+      pageInfo,
+    }).isRequired,
+    onPageChange: func.isRequired,
+    /** A search term to refine the list of notifications */
+    searchTerm: string,
+  };
+
+  state = {
+    sortBy: null,
+    sortDirection: null,
+  };
+
+  createSortedNotifications = memoize(
+    (notificationsConnection, sortBy, sortDirection) => {
+      // if destination coulumn is clicked sort by notificationAddress property
+      const sortByProperty =
+        camelCase(sortBy) === 'destination'
+          ? 'notificationAddress'
+          : camelCase(sortBy);
+
+      if (!sortBy) {
+        return notificationsConnection;
+      }
+
+      return {
+        ...notificationsConnection,
+        edges: [...notificationsConnection.edges].sort((a, b) => {
+          const firstElement =
+            sortDirection === 'desc'
+              ? b.node[sortByProperty]
+              : a.node[sortByProperty];
+          const secondElement =
+            sortDirection === 'desc'
+              ? a.node[sortByProperty]
+              : b.node[sortByProperty];
+
+          return sort(firstElement, secondElement);
+        }),
+      };
+    },
+    {
+      serializer: ([notificationsConnection, sortBy, sortDirection]) => {
+        const ids = sorted(notificationsConnection.edges);
+
+        return `${ids.join('-')}-${sortBy}-${sortDirection}`;
+      },
+    }
+  );
+
+  handleHeaderClick = sortBy => {
+    const toggled = this.state.sortDirection === 'desc' ? 'asc' : 'desc';
+    const sortDirection = this.state.sortBy === sortBy ? toggled : 'desc';
+
+    this.setState({ sortBy, sortDirection });
+  };
+
+  render() {
+    const { classes, onPageChange, notificationsConnection } = this.props;
+    const { sortBy, sortDirection } = this.state;
+    const iconSize = 16;
+    const sortedortedNotificationsConnection = this.createSortedNotifications(
+        notificationsConnection,
+        sortBy,
+        sortDirection
+      );
+
+    return (
+      <ConnectionDataTable
+        connection={sortedortedNotificationsConnection}
+        pageSize={VIEW_DENYLISTED_NOTIFICATIONS_PAGE_SIZE}
+        onHeaderClick={this.handleHeaderClick}
+        onPageChange={onPageChange}
+        headers={tableHeaders}
+        sortByHeader={sortBy}
+        sortDirection={sortDirection}
+        renderRow={({ node }) => (
+          <TableRow key={node.notificationAddress}>
+            <TableCell>
+              <div>
+                <Typography>{node.notificationType}</Typography>
+              </div>
+            </TableCell>
+            <TableCell padding="dense">
+              <div className={classes.listItemCell}>
+                <Typography>{node.notificationAddress}</Typography>
+              </div>
+            </TableCell>
+            <TableCell>
+              <div>
+                <Link to="./">
+                  <DeleteIcon size={iconSize} />
+                </Link>
+              </div>
+            </TableCell>
```

Comment:
The view responsible for creation could also be responsible for deleting. This would remove the need of adding an extra column here. Similar to something like https://taskcluster-web.netlify.com/hooks/garbage/abc456.

Reply:
How will we implement that? 

In [https://taskcluster-web.netlify.com/hooks](https://taskcluster-web.netlify.com/hooks) the delete button appears after we click a hook to view its details but we don't have a details page for the notification entries.

Reply:
In https://taskcluster-web.netlify.com/hooks, the delete page is the same page as the create page. The are both using the `HookForm` component. In our case, we could probably have a `DenyAddressForm` component? and show a delete button if it it does not have a `newDenyAddress` prop. Let me know if that doesn't make sense.

Reply:
Ahh, okay I got what you mean :)